service/history/workflow/mutable_state_impl.go

```diff
@@ -4912,6 +4911,35 @@ func (ms *MutableStateImpl) RecordLastActivityStarted(ai *persistencespb.Activit
 	})
 }
 
+func (ms *MutableStateImpl) RegenerateActivityRetryTask(ai *persistencespb.ActivityInfo) error {
+	// there are two possible cases:
+	//  * this is the first time activity was scheduled
+	//     * in this case we should use current schedule time
+	//	* this is a retry
+	//     * we should recalculate next scheduled time, based on the last time when activity was completed
```

Comment:
What if the activity specified the next retry interval to bypass the retry policy?

Reply:
If you mean "what will happen if retry delay was a part of the failure response" - new scheduled time will be calculated from the new retry policy. If users want to extend retry delay they can do that by specifying new retry policy.

Reply:
I see... so maybe this should be clarified in a comment. Ideally you'll move this whole comment (including my suggestion) into the function's docstring where it'll be more discoverable.

Reply:
the comment clearly state how schedule time will be recalculated. But I also add a note.